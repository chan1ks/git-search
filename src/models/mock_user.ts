import { SearchResponse, Userx } from './search-response.model';
import { User } from './user.model';

export const MOCK_USERSx: Userx[] = [
  {
    login: 'chan1ks',
    id: 9042219,
    node_id: 'MDQ6VXNlcjkwNDIyMTk=',
    avatar_url: 'https://avatars2.githubusercontent.com/u/9042219?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/chan1ks',
    html_url: 'https://github.com/chan1ks',
    followers_url: 'https://api.github.com/users/chan1ks/followers',
    following_url:
      'https://api.github.com/users/chan1ks/following{/other_user}',
    gists_url: 'https://api.github.com/users/chan1ks/gists{/gist_id}',
    starred_url: 'https://api.github.com/users/chan1ks/starred{/owner}{/repo}',
    subscriptions_url: 'https://api.github.com/users/chan1ks/subscriptions',
    organizations_url: 'https://api.github.com/users/chan1ks/orgs',
    repos_url: 'https://api.github.com/users/chan1ks/repos',
    events_url: 'https://api.github.com/users/chan1ks/events{/privacy}',
    received_events_url: 'https://api.github.com/users/chan1ks/received_events',
    type: 'User',
    site_admin: false,
    score: 1.0,
  },
];

export const MOCK_SEARCH_RESPONSE: SearchResponse = {
  total_count: 1,
  incomplete_results: false,
  items: MOCK_USERSx,
};

export const MOCK_USERS: User[] = [
  {
    login: 'can',
    avatar_url: 'https://avatars2.githubusercontent.com/u/357195?v=4',
    html_url: 'https://github.com/can',
    name: 'Can',
    bio: 'Cybersecurity Consultant',
    public_repos: 16,
    followers: 36,
  },
  {
    login: 'candycat1992',
    avatar_url: 'https://avatars3.githubusercontent.com/u/5907064?v=4',
    html_url: 'https://github.com/candycat1992',
    name: 'Feng Lele',
    bio: null,
    public_repos: 12,
    followers: 2280,
  },
  {
    login: 'caniszczyk',
    avatar_url: 'https://avatars1.githubusercontent.com/u/63777?v=4',
    html_url: 'https://github.com/caniszczyk',
    name: 'Chris Aniszczyk',
    bio: 'always paying it forward',
    public_repos: 70,
    followers: 520,
  },
  {
    login: 'cantino',
    avatar_url: 'https://avatars0.githubusercontent.com/u/83835?v=4',
    html_url: 'https://github.com/cantino',
    name: 'Andrew Cantino',
    bio: 'Ruby ◊ JS ◊ Rust ◊ Science ◊ Space ◊ InfoSec ◊ ML ◊ Greentech',
    public_repos: 36,
    followers: 635,
  },
  {
    login: 'burakcan',
    avatar_url: 'https://avatars2.githubusercontent.com/u/3121257?v=4',
    html_url: 'https://github.com/burakcan',
    name: 'Burak Can',
    bio: null,
    public_repos: 49,
    followers: 609,
  },
];
