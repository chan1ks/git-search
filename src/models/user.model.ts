export interface User {
  login?: string;
  name?: string;
  avatar_url?: string;
  html_url?: string;
  bio?: string | null;
  public_repos?: number;
  followers?: number;
}

export interface Users {
  totalCount: number;
  users: User[];
}
