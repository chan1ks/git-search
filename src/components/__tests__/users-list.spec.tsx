import React from 'react';

import renderer from 'react-test-renderer';
import { MOCK_USERS } from '../../models/mock_user';
import UsersList from '../users-list/users-list';

describe('UsersList', () => {
  it('renders correctly', () => {
    const mockedHandleOffsetFn = jest.fn((event) => {
      console.log('Mocked handle offset function');
    });
    const props = {
      users: MOCK_USERS,
      totalCount: MOCK_USERS.length,
      limit: 10,
      handleOffset: mockedHandleOffsetFn,
    };
    const tree = renderer.create(<UsersList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
