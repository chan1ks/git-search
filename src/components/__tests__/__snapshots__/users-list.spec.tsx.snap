// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders correctly 1`] = `
Array [
  <h2>
    5
     github users
  </h2>,
  <ul>
    <li
      className="list-item"
    >
      <a
        className="list-item-link"
        href="https://github.com/can"
      >
        <img
          alt="can"
          className="list-item-avatar"
          src="https://avatars2.githubusercontent.com/u/357195?v=4"
        />
        <div
          className="list-item-body"
        >
          <div
            className="user-details"
          >
            <p
              className="user-call-card"
            >
              <span
                className="bold"
              >
                Can
              </span>
               
              <span>
                can
              </span>
            </p>
            <p>
              Cybersecurity Consultant
            </p>
            <div
              className="flex row"
            >
              <span
                className="light"
              >
                Respos: 
                16
              </span>
              <span
                className="light"
              >
                Followers: 
                36
              </span>
            </div>
          </div>
          <span
            className="chevron right"
          />
        </div>
      </a>
    </li>
    <li
      className="list-item"
    >
      <a
        className="list-item-link"
        href="https://github.com/candycat1992"
      >
        <img
          alt="candycat1992"
          className="list-item-avatar"
          src="https://avatars3.githubusercontent.com/u/5907064?v=4"
        />
        <div
          className="list-item-body"
        >
          <div
            className="user-details"
          >
            <p
              className="user-call-card"
            >
              <span
                className="bold"
              >
                Feng Lele
              </span>
               
              <span>
                candycat1992
              </span>
            </p>
            <p />
            <div
              className="flex row"
            >
              <span
                className="light"
              >
                Respos: 
                12
              </span>
              <span
                className="light"
              >
                Followers: 
                2280
              </span>
            </div>
          </div>
          <span
            className="chevron right"
          />
        </div>
      </a>
    </li>
    <li
      className="list-item"
    >
      <a
        className="list-item-link"
        href="https://github.com/caniszczyk"
      >
        <img
          alt="caniszczyk"
          className="list-item-avatar"
          src="https://avatars1.githubusercontent.com/u/63777?v=4"
        />
        <div
          className="list-item-body"
        >
          <div
            className="user-details"
          >
            <p
              className="user-call-card"
            >
              <span
                className="bold"
              >
                Chris Aniszczyk
              </span>
               
              <span>
                caniszczyk
              </span>
            </p>
            <p>
              always paying it forward
            </p>
            <div
              className="flex row"
            >
              <span
                className="light"
              >
                Respos: 
                70
              </span>
              <span
                className="light"
              >
                Followers: 
                520
              </span>
            </div>
          </div>
          <span
            className="chevron right"
          />
        </div>
      </a>
    </li>
    <li
      className="list-item"
    >
      <a
        className="list-item-link"
        href="https://github.com/cantino"
      >
        <img
          alt="cantino"
          className="list-item-avatar"
          src="https://avatars0.githubusercontent.com/u/83835?v=4"
        />
        <div
          className="list-item-body"
        >
          <div
            className="user-details"
          >
            <p
              className="user-call-card"
            >
              <span
                className="bold"
              >
                Andrew Cantino
              </span>
               
              <span>
                cantino
              </span>
            </p>
            <p>
              Ruby ◊ JS ◊ Rust ◊ Science ◊ Space ◊ InfoSec ◊ ML ◊ Greentech
            </p>
            <div
              className="flex row"
            >
              <span
                className="light"
              >
                Respos: 
                36
              </span>
              <span
                className="light"
              >
                Followers: 
                635
              </span>
            </div>
          </div>
          <span
            className="chevron right"
          />
        </div>
      </a>
    </li>
    <li
      className="list-item"
    >
      <a
        className="list-item-link"
        href="https://github.com/burakcan"
      >
        <img
          alt="burakcan"
          className="list-item-avatar"
          src="https://avatars2.githubusercontent.com/u/3121257?v=4"
        />
        <div
          className="list-item-body"
        >
          <div
            className="user-details"
          >
            <p
              className="user-call-card"
            >
              <span
                className="bold"
              >
                Burak Can
              </span>
               
              <span>
                burakcan
              </span>
            </p>
            <p />
            <div
              className="flex row"
            >
              <span
                className="light"
              >
                Respos: 
                49
              </span>
              <span
                className="light"
              >
                Followers: 
                609
              </span>
            </div>
          </div>
          <span
            className="chevron right"
          />
        </div>
      </a>
    </li>
  </ul>,
  <ul
    className="pagination"
  >
    <li
      className="previous disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        previous
      </a>
    </li>
    <li
      className="next"
    >
      <a
        aria-disabled="false"
        aria-label="Next page"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        next
      </a>
    </li>
  </ul>,
]
`;
